quiver_engine/vis_utils.py

```diff
@@ -7,11 +7,48 @@
 def save_layer_outputs(input_img, model, layer_name, temp_folder, input_path):
 
     with get_evaluation_context():
-        layer_outputs = get_outputs_generator(model, layer_name)(input_img)[0]
 
-        if K.backend() == 'theano':
-            #correct for channel location difference betwen TF and Theano
-            layer_outputs = np.rollaxis(layer_outputs, 0, 3)
+        layer = model.get_layer(layer_name)
+        if ("GlobalAveragePooling2D" in str(type(layer))):
+            next_layer = model.layers[model.layers.index(layer)+1]
+            if ("Dense" in str(type(next_layer))):
+                previous_layer = model.layers[model.layers.index(layer)-1]
+                prev_layer_outputs = get_outputs_generator(model, previous_layer.name)(input_img)[0]
+                next_layer_weights = next_layer.get_weights()[0]
+                prev_layer_channels = next_layer_weights.shape[0]
+                num_classes = next_layer_weights.shape[1]
+                classes_outputs = np.zeros((prev_layer_outputs.shape[0],prev_layer_outputs.shape[1],len(classes)))
+                for cl in range(0,num_classes):
+                    for channel in range(0, prev_layer_channels):
+                        classes_outputs[:, :, cl] = classes_outputs[:, :, cl] + prev_layer_outputs[:,:,channel] * next_layer_weights[channel,cl]
+
+                classes_outputs = deprocess_image(classes_outputs)
+                return [save_layer_img(
```

Comment:
If we want them not to be in conflict with others for the same layer - yes. I can do that in a spare moment.